pact/fixed-quote-policy-royalty.pact

```diff
@@ -0,0 +1,192 @@
+(namespace (read-msg 'ns))
+
+(module fixed-quote-policy-royalty GOVERNANCE
+
+  @doc "Policy for fixed issuance with simple quoted sale."
+
+  (defcap GOVERNANCE ()
+    (enforce-guard (keyset-ref-guard 'marmalade-ns-admin )))
+
+  (implements kip.token-policy-v1_DRAFT3)
+  (use kip.token-policy-v1_DRAFT3 [token-info])
+
+  (defschema policy-schema
+    fungible:module{fungible-v2}
+    creator:string
+    creator-guard:guard
+    mint-guard:guard
+    max-supply:decimal
+    min-amount:decimal
+    royalty-rate:decimal
+  )
+
+  (deftable policies:{policy-schema})
+
+  (defconst QUOTE "quote"
+    @doc "Payload field for quote spec")
+
+  (defschema quote-spec
+    @doc "Quote data to include in payload"
+    price:decimal
+    recipient:string
+    recipient-guard:guard
+    )
+
+  (defschema quote-schema
+    id:string
+    spec:object{quote-spec})
+
+  (deftable quotes:{quote-schema})
+
+  (defun get-policy:object{policy-schema} (token:object{token-info})
+    (read policies (at 'id token))
+  )
+
+  (defun enforce-ledger:bool ()
+     (enforce-guard (marmalade.ledger.ledger-guard))
+   )
+
+  (defun enforce-mint:bool
+    ( token:object{token-info}
+      account:string
+      amount:decimal
+    )
+    (enforce-ledger)
+    (bind (get-policy token)
+      { 'mint-guard:=mint-guard:guard
+      , 'max-supply:=max-supply:decimal
+      }
+      (enforce-guard (at 'mint-guard (get-policy token)))
+      (enforce (<= (+ amount (at 'supply token)) max-supply) "Exceeds max supply")
+  ))
+
+  (defun enforce-burn:bool
+    ( token:object{token-info}
+      account:string
+      amount:decimal
+    )
+    (enforce-ledger)
+    (enforce false "Burn prohibited")
+  )
+
+  (defun enforce-init:bool
+    ( token:object{token-info}
+    )
+    (enforce-ledger)
+    (let* ( (fungible:module{fungible-v2} (read-msg 'fungible ))
+            (creator:string (read-msg 'creator ))
+            (creator-guard:guard (read-keyset 'creator-guard ))
+            (mint-guard:guard (read-keyset 'mint-guard ))
+            (max-supply:decimal (read-decimal 'max-supply ))
+            (min-amount:decimal (read-decimal 'min-amount ))
+            (royalty-rate:decimal (read-decimal 'royalty-rate ))
+            (creator-details:object (fungible::details creator ))
+            )
+      (fungible::precision)
+      (enforce (=
+        (at 'guard creator-details) creator-guard)
+        "Creator guard does not match")
+      (enforce (and
+        (>= royalty-rate 0.0) (<= royalty-rate 1.0))
+        "royalty rate is not valid")
```

Comment:
```suggestion
        "Invalid royalty rate")
```
(avoid "is not" in error messages when possible, it's shorter and less conversational)